packages/fab/src/test/mwc-fab.test.ts

```diff
@@ -89,4 +89,66 @@ suite('mwc-fab', () => {
         button = element.shadowRoot!.querySelector('button')!;
         assert.equal(button.getAttribute('aria-label'), 'check');
       });
+
+  test('setting `mini` sets the correct inner class', async () => {
+    await element.updateComplete;
+    const miniClass = 'mdc-fab--mini';
+    const button = element.shadowRoot!.querySelector('.mdc-fab')!;
+    assert.isFalse(button.classList.contains(miniClass));
+    element.mini = true;
+    await element.updateComplete;
+    assert.isTrue(button.classList.contains(miniClass));
+  });
+
+  test('setting `exited` sets the correct inner class', async () => {
+    await element.updateComplete;
+    const exitedClass = 'mdc-fab--exited';
+    const button = element.shadowRoot!.querySelector('.mdc-fab')!;
+    assert.isFalse(button.classList.contains(exitedClass));
+    element.exited = true;
+    await element.updateComplete;
+    assert.isTrue(button.classList.contains(exitedClass));
+  });
+
+  test('setting `extended` sets the correct inner class', async () => {
+    await element.updateComplete;
+    const extendedClass = 'mdc-fab--extended';
+    const button = element.shadowRoot!.querySelector('.mdc-fab')!;
+    assert.isFalse(button.classList.contains(extendedClass));
+    element.extended = true;
+    await element.updateComplete;
+    assert.isTrue(button.classList.contains(extendedClass));
+  });
+
+  test('displays label only if extended', async () => {
+    await element.updateComplete;
+    let label = element.shadowRoot!.querySelector(LABEL_SELECTOR);
+    assert.equal(label, null);
+    element.label = 'foo';
+    await element.updateComplete;
+    label = element.shadowRoot!.querySelector(LABEL_SELECTOR);
+    assert.equal(label, null);
+    element.extended = true;
+    await element.updateComplete;
+    label = element.shadowRoot!.querySelector(LABEL_SELECTOR);
+    assert.instanceOf(label, Element);
+    assert.equal(label!.textContent, 'foo');
+  });
+
+  test('`showIconAtEnd` displays icon after label', async () => {
+    element.icon = 'check';
+    element.label = 'foo';
+    element.extended = true;
+    element.showIconAtEnd = false;
+    await element.updateComplete;
+
+    let node = element.shadowRoot!.querySelector(
+        `${ICON_SELECTOR} + ${LABEL_SELECTOR}`)!;
```

Comment:
gross selector is gross, but I'll accept it

Reply:
Hahaha. I was hoping there would be some kind of "is after" built in on Node. But apparently not. 

Alternatively we can check the index in `children` of the parent or that nextSibling is the label

Reply:
this is fine; it's effectively the same thing and better than nextSibling since it might pick up a text node. Children also works